lib/AST/LookupVisibleDecls.cpp

```diff
@@ -695,7 +695,7 @@ static void lookupVisibleMemberDeclsImpl(
     // If we have a class type, look into its superclass.
     auto *CurClass = dyn_cast<ClassDecl>(CurNominal);
 
-    if (CurClass && CurClass->hasSuperclass()) {
+    if (CurClass && CurClass->getSuperclass()) {
```

Comment:
Should this really be a getSuperclass()? If so, it warrants a comment explaining why the next person to visit this shouldn't just change it back to hasSuperclass().

Reply:
It currently needs to be `getSuperclass()`, because this code wants the `Type` and not the `ClassDecl`. All of this code needs to be updated to use `Decl`-based lookup rather than `Type`-based lookup, to reduce the number of dependencies. I'm going to add a big ugly comment and may very well eliminate `hasSuperclass()` or rename it to `hasSuperclassDecl()`.